docs/build/basics/scripts.md

```diff
@@ -87,6 +87,19 @@ flow scripts execute ./helloWorld.cdc
 
 A user can define their own scripts or can use already defined scripts by the contract authors that can be found by using the [FLIX](../../tools/flow-cli/flix) service.
 
+## Best Practices
+
+Following are some recommendations on how to write efficient scripts:
+
+1. **Simpler and shorter scripts**: Scripts, like transactions, are subject to computation limits (see [limitations](#limitations)). It is recommended to run shorter and simpler scripts which have low time complexity for a faster response. If you have a script with several nested loops or long iteration, consider simplifying the script logic.
+
+
+2. **Smaller length of array or dictionary type arguments**: If your script requires an array or a dictionary as an argument, instead of making a single script call with a long list, make multiple calls with a smaller subset of the array or dictionary.
```

Comment:
The slowness is less about the input data, and more about the number of registers touched. large input lists imply the script will read many accounts/fields.

Reply:
correct - hence I am suggesting to pass in smaller list - this is just the same suggestion of having low time complexity as 1 but i have seen examples of scripts which pass in a huge number of params e.g. the discord bot by emrald dao which pass in a long list of Flow addresses it wants to verify. That script sped up significantly if I reduced the arg list by 50%.

Reply:
I am assuming the target audience for this is not necessary developers but also perhaps folks who are just getting started.